### ISSUE TITLE
FileEditor dialog inconsistently opens with PyQt5 5.15 on macOS

### ISSUE THREAD

**Issue by yoda-vid** [enthought/traitsui#1364]
I have found the `FileEditor` dialog window to sporadically fail to open starting with PyQ5 5.15 on macOS. Sometimes the dialog will open on the first try, other times I will need to click it up to 20 times to open the dialog. The behavior is also inconsistent with consecutive button presses. A file dialog opened through PyFace does not have this issue.

Testing:
I ran the File Editor demo as follows and clicked on the folder button in the `File` widget:
```
python traitsui/examples/demo/Standard_Editors/File_Dialog/File_Open.py
```

**Edit (2020-10-19)**: I posted the incorrect demo, which should be:
```
python traitsui/examples/demo/Standard_Editors/FileEditor_demo.py
```

For comparison with PyFace, I ran its Python Editor demo and clicked on the `Open` button:
```
python examples/python_editor.py
```

Environment:
- Create a barren Venv environment with Python 3.6
- Pip install traitsui and PyQt5 (current version is 5.15.1)
- Test as above
- Pip uninstall PyQt5 and install PyQt5==5.14.2
- Test as above

I encountered this behavior in macOS 10.14 and 10.15 but did not have any issues on Windows 10 (there I used the Conda traitsui package). When testing on prior versions of PyQt5 (eg 5.14.2), the dialog opens consistently.

**Comment by kitchoi** [comment]
Hi @yoda-vid, thank you for your report!

I am testing this on macOS 10.15.5, and this is what I did:
```
$ python3.8 -m venv my_env
$ source my_env/bin/activate
(my_env) $ pip install traitsui pygments pyqt5==5.15.1
(my_env) $ python traitsui/examples/demo/Standard_Editors/File_Dialog/File_Open.py
```

I tried to press the button many times, rerunning the script, but I could not reproduce the issue: the file dialog opens reliably.
I also tried installing `pyqt5==5.14.2`, and I could not reproduce.
Perhaps there is something I miss?

**Comment by yoda-vid** [comment]
Thanks for testing, @kitchoi !

I realized that I posted the incorrect demo, which should be the `FileEditor_demo` instead. I followed your approach except ran this example:

```
python traitsui/examples/demo/Standard_Editors/FileEditor_demo.py
```

I retested PyQt5 5.15.1 and continued to get sporadic failures, whereas I had no problems with 5.14.2. With PyQt5 5.15.1, I can get runs of 5-6 successful openings, then 1 failure, 5 more successes, 3 failures, etc. Record so far is 56 clicks without opening. Tried clicking on different parts of the icon but couldn't find a correlation.

Apologies for reporting the incorrect demo, which I will fix in the original post.

**Comment by kitchoi** [comment]
Thank you @yoda-vid for your updates!

I was able to reproduce and the issue is now fixed by #1383.
However since the fix is made so close the upcoming release 7.1.0 (the release candidate was out last week), the fix did not make it into 7.1.0. It will probably have to be included in the next feature or bugfix release.

Thanks again for reporting the issue.

**Comment by yoda-vid** [comment]
Thanks @kitchoi , this is really awesome!

I can confirm that the fix works great for me on the master branch with both PyQt5 5.14.2 and 5.15.1. I hadn't thought it had to do with the click location, but clearly it does! Thank you again for finding it and for the fix.
